src/app/clusters/door-lock-server/door-lock-server.cpp

```diff
@@ -718,6 +715,15 @@ void DoorLockServer::getCredentialStatusCommandHandler(chip::app::CommandHandler
         }
     }
 
+    sendGetCredentialResponse(commandObj, commandPath, credentialType, credentialIndex, userIndexWithCredential, credentialInfo,
+                              credentialExists);
+}
+
+void DoorLockServer::sendGetCredentialResponse(chip::app::CommandHandler * commandObj,
+                                               const chip::app::ConcreteCommandPath & commandPath, DlCredentialType credentialType,
+                                               uint16_t credentialIndex, uint16_t userIndexWithCredential,
+                                               EmberAfPluginDoorLockCredentialInfo credentialInfo, bool credentialExists)
```

Comment:
Please take a look at my comment above about making this parameter either `Optional` or a regular pointer. I think we need to do that anyway as copying the whole structure is a bit expensive.

Reply:
That's a good point. I think a pointer would work well here. Sending a null for the error cases i.e. when credentialIndex is invalid sounds good to me. I'll look into the implications of making it optional as well. I'll update this, thank you.

Reply:
I changed the function signature to take a pointer, and am now passing null in the error case.